### PR TITLE
chore: add exhaustivness error utility

### PR DESCRIPTION
**Pull request by EvanHahn**
We have a few exhaustiveness checks. I think it's time to DRY these out into a simple utility function.

(I plan to use this in an upcoming change, which is why I'm doing this now.)

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -66,6 +66,13 @@ export async function openedNoiseSecretStream(stream) {
   return /** @type {OpenedNoiseStream | DestroyedNoiseStream} */ (stream)
 }
 
+/**
+ * @param {never} value
+ * @returns {Error}
+ */
+export const exhaustivenessError = (value) =>
+  new Error(`Exhaustiveness check failed. ${value} should be impossible`)
```
Comment (achou11) [inline src/utils.js:74]: thoughts on an implementation that extends `Error` instead? e.g.

```js
class ExhaustivenessError extends Error {
  constructor(value) {
    this.name = 'ExhaustivenessError'
    this.message = `Exhaustiveness check failed. ${value} should be impossible`
  }
}
```

File: tests/utils.js [hunk 2]
```diff
@@ -1,12 +1,25 @@
...
+  bools.forEach((bool) => {
+    switch (bool) {
+      case true:
+      case false:
+        break
+      default:
+        throw exhaustivenessError(bool)
+    }
+  })
+})
```
[the remark this answers is not in the document]
Comment (achou11) [reply tests/utils.js:21]: you can use [`t.execution()`](https://github.com/holepunchto/brittle?tab=readme-ov-file#texecutionpromisefunctionasync-function-message)

**Review by achou11 — APPROVED** [review]
Couple of non-blocking ideas/suggestions but otherwise fine with the addition

File: tests/utils.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (EvanHahn) [reply tests/utils.js]: Thanks! Done in c83a1209262d4092d3afafc6c7455accea7b1bdf.

**Comment by gmaclennan** [comment]
The purpose of these exhaustiveness checks is to get a typescript error if we are not checking all the options. Does this achieve the same thing?

**Comment by gmaclennan** [comment]
If it does then this looks good to me.

**Comment by EvanHahn** [comment]
> The purpose of these exhaustiveness checks is to get a typescript error if we are not checking all the options. Does this achieve the same thing?

Yup! Exact same thing.

**Pull request merged**